### PR TITLE
Refactor Proxy as an interface to allow override impl

### PR DESCRIPTION
**Pull request by yongjiapro**
**What this PR does**:

Refactor proxy.Proxy as an interface to allow override impl

修改proxy.Proxy为接口类型，允许项目在自定义ProxyFactory时自定义实现proxy

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

在网关泛调用场景下，当前GenericService实现需要修改对返回数据的处理逻辑。
目前ProxyFactory允许自定义注册，但其创建的proxy.Proxy无法重新实现。

**Does this PR introduce a user-facing change?**:

```release-note
Rft: refactor proxy.Proxy as an interface
```

### CODE REVIEW

**Comment by cityiron** [comment]
I want to know more, ps leave a contact information, if you use Dingding, our Group is 31363295.

**Comment by AlexStocks** [comment]
@yongjiapro hey, guy, give u two suggestions.

1  the target branch of this pr should be develop, not the master branch.
2 U can define a interface whose name maybe ProxyInterface or Interface in k8s style.

File: common/proxy/proxy.go [hunk 1]
```diff
@@ -36,8 +36,17 @@ import (
 	invocation_impl "github.com/apache/dubbo-go/protocol/invocation"
 )
 
+type Proxy interface {
```
Comment (AlexStocks) [inline common/proxy/proxy.go:39]: ProxyInterface or Interface in k8s style

File: common/proxy/proxy.go [hunk 2]
```diff
@@ -36,8 +36,17 @@ import (
...
+	Get() common.RPCService
+	GetCallback() interface{}
+	GetInvoker() protocol.Invoker
+}
+
+var _ Proxy = new(DefaultProxy)
+
 // nolint
-type Proxy struct {
+type DefaultProxy struct {
```
Comment (AlexStocks) [inline common/proxy/proxy.go:49]: Pls do not change the existing struct name.

**Comment by yongjiapro** [comment]
Force update to develop branch, and:
1. Update Proxy to ProxyInterface;
2. Restore Proxy struct

**Comment by yongjiapro** [comment]
> I want to know more, ps leave a contact information, if you use Dingding, our Group is 31363295.

已发送入群申请

**Comment by yongjiapro** [comment]
Re-pr a new impl

**Pull request closed without merging**